File: Report.Application/Dto/Include/FilterParameter.cs
using Microsoft.VisualBasic;
using Report.Application.Enum;
using Report.Domain;

namespace Report.Application.Dto.Include
{
    
    public class FilterParameter
    {
        public int? AssisttanceType { get; set; }
        public int? ManagementType { get; set; }
        public int? HallType { get; set; }
        public string HallCode { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public Meter Meter { get; set; }     
        public Period Period { get; set; }
    }
}

File: client-app/src/app/Dto/Exclude/FilterParameter.ts
import { unitModel } from "src/app/models/unitModel";

export interface filterParameter {
    assisttanceType: number | undefined;
    managementType: number | undefined;
    hallType: number | undefined;
    hallCode: string | undefined;
    startDate: string | null;
    endDate: string | null;
    meter: Meter | undefined;
    period: Period | undefined;
}

export enum Meter {
    Water = 1,
    Gas = 2,
    CompresAir = 4,
    Electricity = 5,
}

export enum Period {
    Minute = 0,
    Hour = 1,
    Day = 2,
    Month = 3,
}